src/Symfony/Component/Serializer/Deserialize/DataModel/DataModelBuilder.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\DataModel;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Serializer\Deserialize\Config\DeserializeConfig;
+use Symfony\Component\Serializer\Deserialize\Mapping\PropertyMetadataLoaderInterface;
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Php\ArgumentsNode;
+use Symfony\Component\Serializer\Php\FunctionCallNode;
+use Symfony\Component\Serializer\Php\MethodCallNode;
+use Symfony\Component\Serializer\Php\PhpNodeInterface;
+use Symfony\Component\Serializer\Php\ScalarNode as PhpScalarNode;
+use Symfony\Component\Serializer\Php\VariableNode;
+use Symfony\Component\Serializer\Type\Type;
+use Symfony\Component\VarExporter\ProxyHelper;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class DataModelBuilder implements DataModelBuilderInterface
+{
+    public function __construct(
+        private readonly PropertyMetadataLoaderInterface $propertyMetadataLoader,
+        private readonly ContainerInterface $runtimeServices,
```

Comment:
```suggestion
        private readonly ?ContainerInterface $runtimeServices = null,
```

Make this argument optional to ease the usage of the standalone component.